extends Node

onready var globals = $"/root/Globalnode"

onready var titlemusic = $TitleMusicPlayer
onready var music = $MusicPlayer

var finalscore = 0
var highscore = 1
var ammo = 50
var health = 100
var coins = 0
var gamerows = 20
var music_volume = 0
var effects_volume = -6

func _ready():
	titlemusic.volume_db = globals.music_volume
	load_game()
	print(highscore)	

	
func load_game():
	var load_game = File.new()
	if not load_game.file_exists("user://savegame.save"):
		return # Error! We don't have a save to load.

	# We need to revert the game state so we're not cloning objects
	# during loading. This will vary wildly depending on the needs of a
	# project, so take care with this step.
	# For our example, we will accomplish this by deleting saveable objects.
	var load_nodes = get_tree().get_nodes_in_group("persist")
	for i in load_nodes:
		i.queue_free()

	# Load the file line by line and process that dictionary to restore
	# the object it represents.
	load_game.open("user://savegame.save", File.READ)
	while load_game.get_position() < load_game.get_len():
		# Get the saved dictionary from the next line in the save file
		var node_data = parse_json(load_game.get_line())
		print("Node Data: ", node_data)
		if node_data.has("effectsvolume"):
			#music_volume = node_data["musicvolume"]
			effects_volume = node_data["effectsvolume"]
		elif node_data.has("highscore"):
			highscore = node_data["highscore"]
		elif node_data.has("coins"):
			coins = node_data["coins"]
		# Firstly, we need to create the object and add it to the tree and set its position.
#		var new_object = load(node_data["filename"]).instance()
#		get_node(node_data["parent"]).add_child(new_object)
#		new_object.position = Vector2(node_data["pos_x"], node_data["pos_y"])

		# Now we set the remaining variables.
#		for i in node_data.keys():
#			if i == "filename" or i == "parent" or i == "pos_x" or i == "pos_y":
#				continue
#			new_object.set(i, node_data[i])
	load_game.close()
	
func save_game():
	var save_game = File.new()
	save_game.open("user://savegame.save", File.WRITE)
	var save_nodes = get_tree().get_nodes_in_group("persist")
	for node in save_nodes:
		print("node  ",save_nodes)
		# Check the node is an instanced scene so it can be instanced again during load
#		if node.filename.empty():
#			print("persistent node '%s' is not an instanced scene, skipped" % node.name)
#			continue

		# Check the node has a save function
		if !node.has_method("save"):
			print("persistent node '%s' is missing a save() function, skipped" % node.name)
			continue

		# Call the node's save function
		var node_data = node.call("save")
		# Store the save dictionary as a new line in the save file
		save_game.store_line(to_json(node_data))
	save_game.close()	


